proto/gnmi_ext/gnmi_ext.proto

```diff
@@ -28,12 +28,13 @@ option go_package = "github.com/openconfig/gnmi/proto/gnmi_ext";
 // The Extension message contains a single gNMI extension.
 message Extension {
   oneof ext {
-    RegisteredExtension registered_ext = 1;  // A registered extension.
+    RegisteredExtension registered_ext = 1;     // A registered extension.
     // Well known extensions.
-    MasterArbitration master_arbitration = 2;  // Master arbitration extension.
-    History history = 3;                       // History extension.
-    Commit commit = 4;                         // Commit confirmed extension.
-    Depth depth = 5;                           // Depth extension.
+    MasterArbitration master_arbitration = 2;   // Master arbitration extension.
+    History history = 3;                        // History extension.
+    Commit commit = 4;                          // Commit confirmed extension.
+    Depth depth = 5;                            // Depth extension.
+    ConfigSubscription config_subscription = 6; // Config Subscription extension.
```

Comment:
Hi @robshakir @dplore, all
 
We would like to resume the discussions around this extension proposal.

After extensive internal use of this extension we find it particularly fit for a common problem in the network management domain -- **configuration deviation/drift detection and reconciliation**. We believe it might be useful for operators and other gNMI users.

To summarize, ConfigSubscription extension provides the following capabilities that contribute to the config drift detection use case:
1. An ability to **subscribe to configuration only data**
    Contrary to Get RPC, Subscribe RPC does not have a toggle to select which category of data nodes to subscribe to (state vs config vs both)
    This is important for gNMI applications that are based on yang models without strict state/config separation of data
2. A way to **identify commit boundaries** when subscribed to configuration data.
    The `sync_done` message allows a client to identify when all changes for a particular commit have been fully sent by the gNMI server. This ensures it is safe to trigger deviation computation without needing to perform a full resync.


These two new capabilities added by this extension to gNMI could (among other use cases) enable a gNMI client to make use of a robust and reactive configuration drift detection/reconciliation mechanism.  
When used together, the configuration mananagement system can make use of the streaming nature of gNMI Subscribe RPC to get a rapid deviation detection.

First, it benefits from a **lightweight** subscription mode when only changes to the configuration datastore elements are being sent as notifications.

And then, by making use of the `sync_done` message in this extension the system can quickly react to the changes made to the configuration data it is subscribed to.

Reply:
I think the use case makes sense, we can probably stop selling it and work out the details :-)

The subscription to config only data is fine -- I would be OK with adding something here to `SubscriptionList` that says the the data type similarly to `Get`. We can just make the default state of this as `BOTH`, which would give a means to be able to be backwards compatible should this field not be specified.